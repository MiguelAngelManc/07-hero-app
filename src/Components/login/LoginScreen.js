import React, { useContext } from 'react'
import { AuthContext } from '../../auth/AuthContext'
import { useForm } from '../../hooks/useForm'
import { types } from '../../types/types'

export const LoginScreen = ( {history} ) => {

    const {user, dispatch} = useContext(AuthContext)


    const lastName = localStorage.getItem('lastname') || ''

    const [{username}, handleInputChange] = useForm({username:lastName})

    const handleLogin = () =>{

        const lastpath = localStorage.getItem('lastpath') || '/'

        const action = {
            type:types.login,
            payload:{
                name:username
            }
        }

        console.log(username)

        console.log(action)

        dispatch(action)

        history.replace(lastpath)

    }


    console.log(user)


    return (
        <div className="container mt-5">
            <h1>Login</h1>
            <hr />

            <input 
                type="text"
                name="username"
                value={username}
                onChange={handleInputChange}
                autoComplete="off"
                placeholder="Username"
            />
            <button 
                className="btn btn-primary btn-box"
                onClick={()=>{handleLogin()}}>
                    Login!
            </button>


        </div>
    )
}
