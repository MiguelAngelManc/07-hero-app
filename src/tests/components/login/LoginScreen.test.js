import { mount } from "enzyme"
import { AuthContext } from "../../../auth/AuthContext"
import { LoginScreen } from "../../../Components/login/LoginScreen"
import { types } from "../../../types/types"

describe('Pruebas en LoginScreen', () => {
    
    const history = {
        replace: jest.fn(),
    }

    const contextMock = {
        user:{},
        dispatch: jest.fn(),
    }

    test('Debe mostrarse correctamente', () => {
        
        const wrapper = mount(
            <AuthContext.Provider value={contextMock}>
                <LoginScreen history={history}/>
            </AuthContext.Provider>
        )

        expect(wrapper).toMatchSnapshot();
    })

    test('Debe hacer login y cambiar el usuario y la navegacion', () => {

        const wrapper = mount(
            <AuthContext.Provider value={contextMock}>
                <LoginScreen history={history}/>
            </AuthContext.Provider>
        )

        const handleClick = wrapper.find('button').prop('onClick');

        const action = {
            type:types.login,
            payload:{
                name:''
            }
        }

        handleClick()

        expect(history.replace).toHaveBeenCalledWith('/')
        expect(contextMock.dispatch).toHaveBeenCalledWith(action)

        localStorage.setItem('lastpath','/marvel')
        localStorage.setItem('lastname','Miguel')

        handleClick()

        
        expect(history.replace).toHaveBeenCalledWith('/marvel')

    })
    
    

})
