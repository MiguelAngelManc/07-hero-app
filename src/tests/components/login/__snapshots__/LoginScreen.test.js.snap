// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en LoginScreen Debe mostrarse correctamente 1`] = `
<div
  className="container mt-5"
>
  <h1>
    Login
  </h1>
  <hr />
  <input
    autoComplete="off"
    name="username"
    onChange={[Function]}
    placeholder="Username"
    type="text"
    value=""
  />
  <button
    className="btn btn-primary btn-box"
    onClick={[Function]}
  >
    Login!
  </button>
</div>
`;
